### ISSUE TITLE
The `circular` attribute is redundant to crosslinking

### ISSUE THREAD

**Issue by paulflang** [KarrLab/bpforms#109]
The `circular` attribute seems redundant to crosslinking (at least for proteins). Not sure if we are OK with providing two ways to do the same thing.

**Comment by jonrkarr** [comment]
Semantically, I think its useful to have a special syntactic sugar for circularity. Semantically, this connotes different meaning than a crosslink.

More generally, BpForms is not founded on the principle that there should only be 1 unique way to describe a polymer.
- For circular polymers, there isn't necessarily a well-defined start position
- There's nothing that prevents the community from defining monomeric forms which are themselves polymers. If such a monomeric form is semantically meaningful on its own and has a commonly accepted id or name, I don't see a problem with this.

The main goal of BpForms is to be concrete. I don't think this requires a unique representation of each molecule.

**Comment by paulflang** [comment]
OK, thanks. This all makes sense.

> For circular polymers, there isn't necessarily a well-defined start position

I noticed that this a problem where the bpforms equality check fails. I had a brief look at the code, but the way it is structured I couldn't find a trivial fix.

**Comment by jonrkarr** [comment]
Yes, that a good point. We should change the code to recognize the equivalence of this special case. This comes up in a couple places
- equality
- it shouldn't be possible to describe a circular molecule with a crosslink between the same atoms; this should result in an error.
